vision/diffusion_mnist/diffusion_mnist.jl

```diff
@@ -0,0 +1,385 @@
+# Score-Based Generative Modeling
+#
+# Score-Based Generative Modeling through Stochastic Differential Equations.
+# Yang Song, Jascha Sohl-Dickstein, Diederik P. Kingma, 
+# Abhishek Kumar, Stefano Ermon, and Ben Poole.
+# https://arxiv.org/pdf/2011.13456.pdf
+
+using MLDatasets
+using Flux
+using Flux: @functor, chunk, params
+using Flux.Data: DataLoader
+using Parameters: @with_kw
+using BSON
+using CUDA
+using DrWatson: struct2dict
```

Comment:
If we write a simple utility, we can avoid a package dependency here. A separate tutorial showing project tracking for Flux with DrWatson would be an awesome addition!

Reply:
What is meant by project tracking? Would the simple dependency just be converting a struct to a dict so that we can serialize with `BSON`? I was trying to keep my already long PR as short as possible, but if you think adding a separate utility to accomplish this is of value, I would be happy to take a look 😃

Reply:
Much clearer, thanks!